tweet_recommendations/estimator.py

```diff
@@ -0,0 +1,26 @@
+from typing import *
+
+import pandas as pd
+from sklearn.base import TransformerMixin, BaseEstimator
+
+
+class Estimator(TransformerMixin, BaseEstimator):
+    def fit(self, x: pd.DataFrame, y: Optional[pd.DataFrame] = None,
+            **fit_params) -> "Estimator":
+        raise NotImplementedError
+
+    def transform(self, x: List[str]) -> List[str]:
+        raise NotImplementedError
+
+    def drop_tweets_with_hashtags_that_occurred_less_than(self, data: pd.DataFrame,
```

Comment:
Ta metoda:
- jest statyczna (mogłaby być po prostu funkcją)
- filtruje hashtagi, które pojawiają się rzadziej, niż określa to parametr
- usuwa tweety, które nie mają hashtagów

IMO należałoby wydzielić z tej metody dwie funkcje, realizujące pojedyncze zadania

Reply:
2) i 3) w porządku, natomiast 1) mam mieszane uczucia. Taka metoda jest specyficzna dla konkretnego zastosowania, nie ma sensu jej otwierać na zewnątrz. Generalnie powinna być protected, ale nie ma tego w pythonie ;)

Reply:
W Pythonie koncepcja OOP jest zdecydowanie luźniejsza, niż np. w Javie, dlatego IMO wydzielenie tego w funkcję (lub dwie osobne funkcje) jest OK. Tutaj [przykład](https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/cluster/optics_.py) z samego sklearna - funkcje pomocnicze wydzielone są poza klasę.

Reply:
Zwróciłbym tu przy okazji uwagę, że ta funkcja jest już gdzieś na zewnątrz zaimplementowana i teraz raczej duplikuje się kod